docs/source/conf.py

```diff
@@ -1,84 +1,291 @@
+    'sphinx.ext.autosectionlabel',
 ]
+
+
+intersphinx_mapping = {
+    'torch': ('https://pytorch.org/docs/stable/', None),
+    'numpy': ('https://numpy.org/doc/stable/', None),
+    'ismrmrd': ('https://ismrmrd.readthedocs.io/en/latest/', None),
+    'einops': ('https://einops.rocks/', None),
+    'python': ('https://docs.python.org/3', None),
+    'pydicom': ('https://pydicom.github.io/pydicom/stable/', None),
+    'pypulseq': ('https://pypulseq.readthedocs.io/en/master/', None),
+    'torchkbnufft': ('https://torchkbnufft.readthedocs.io/en/stable/', None),
+    'scipy': ('https://docs.scipy.org/doc/scipy/', None),
+    'ptwt': ('https://pytorch-wavelet-toolbox.readthedocs.io/en/latest/', None),
+    'typing-extensions': ('https://typing-extensions.readthedocs.io/en/latest/', None),
+}
+
+autosectionlabel_prefix_document = True
+napoleon_use_param = True
+napoleon_use_rtype = False
+typehints_defaults = 'comma'
+typehints_use_signature = True
+typehints_use_signature_return = True
+typehints_use_rtype = False
+typehints_document_rtype = False
 autosummary_generate = True
 autosummary_imported_members = False
 autosummary_ignore_module_all = False
 autodoc_member_order = 'groupwise'
+autodoc_preserve_defaults = True
+autodoc_class_signature = 'separated'
 templates_path = ['_templates']
 exclude_patterns = ['_build', 'Thumbs.db', '.DS_Store']
 source_suffix = {'.rst': 'restructuredtext', '.txt': 'restructuredtext', '.md': 'markdown'}
-
 myst_enable_extensions = [
-    "amsmath",
-    "dollarmath",
+    'amsmath',
+    'dollarmath',
 ]
-nb_execution_mode = "off"
-
-# -- Options for HTML output -------------------------------------------------
-# https://www.sphinx-doc.org/en/master/usage/configuration.html#options-for-html-output
-
+nb_execution_mode = 'auto'
+nb_output_stderr = 'remove'
+nb_output_stdout = 'remove'
+nb_execution_timeout = 120
 html_theme = 'sphinx_rtd_theme'
 html_title = name
 html_show_sphinx = False
+html_show_sourcelink = False
 html_static_path = ['_static']
 html_css_files = ['custom.css']
 html_logo = '_static/logo_white.svg'
 html_sidebars = {'**': ['search-field', 'sidebar-nav-bs']}
 html_theme_options = {
     'logo_only': True,
-    'pygment_light_style': 'default',
-    'pygment_dark_style': 'github-dark',
-    'show_toc_level': 3,
-    'icon_links': [
-        {
-            # Label for this link
-            'name': 'GitHub',
-            # URL where the link will redirect
-            'url': 'https://github.com/PTB-MR/mrpro',
-            # Icon class (if "type": "fontawesome"), or path to local image (if "type": "local")
-            'icon': 'fa-brands fa-github',
-        },
-    ],
+    'collapse_navigation': False,
 }
+html_context = {
+    'display_github': False,
+    'github_user': 'PTB-MR',
+    'github_repo': 'mrpro',
+    'github_version': 'main',
+}
+linkcode_blob = html_context['github_version']
+linkcode_link_text = '[source]'
+default_role = 'py:obj'
+
+
+def get_lambda_source(obj):
```

Comment:
used to supports `default_factory=lambda: torch.ones(1,1,1,1,1)` etc in `rewrite_dataclass_init_default_factories`